lib/RGBot.js

```diff
@@ -248,43 +473,75 @@ const RGBot = class {
+     * <i><b>Experimental - The behaviour of this API can and almost certainly will change in a future API version.</b></i>
+     *
+     * Determines whether an Entity's username or displayName or name are equal to a targetName string.
+     * Matching for username and name is case-sensitive.  Matching for displayName is NOT case-sensitive.
+     * Generally you should use entityNamesMatch(targetName, entity, options) instead of this function.
+     *
+     * @param {string} targetName
+     * @param {Entity|Item} entity
+     * @param {object} [options={}] Optional parameters
+     * @param {boolean} [options.partialMatch=false] Allow partial matches. For example, '_planks' will match any Entity containing 'planks' in its name ('spruce_planks', 'oak_planks', etc.)
+     * @returns {boolean}
+     *
+     * @example <caption>Full Match</caption>
+     * const entity = rg.getItemDefinitionByName('iron_axe')
+     * rgBot.entityDisplayNamesMatch('iron_axe', entity) // returns true - matched name
+     * rgBot.entityDisplayNamesMatch('Iron Axe', entity) // returns true - matched displayName
+     * rgBot.entityDisplayNamesMatch('iron axe', entity) // returns true - matched displayName
+     *
+     * @example <caption>Partial Match</caption>
+     * const entity = rg.getItemDefinitionByName('iron_axe')
+     * rgBot.entityDisplayNamesMatch('_axe', entity, {partialMatch: true}) // returns true
+     */
+    entityDisplayNamesMatch(targetName, entity, options = {}) {
+        const partialMatch = options.partialMatch || false;
+        const usernameMatch = (username, targetName) => { return username && ((username === targetName) || (partialMatch && username.includes(targetName)))};
+        const namesMatch = (entityName, targetName) =>  { return entityName && ((entityName === targetName) || (partialMatch && entityName.includes(targetName)))};
+        const displayNamesMatch = (entityDisplayNameLower, targetNameLower) => { return entityDisplayNameLower && ((entityDisplayNameLower === targetNameLower) || (partialMatch && entityDisplayNameLower.includes(targetNameLower)))};
+        return usernameMatch(entity.username, targetName) || displayNamesMatch(entity.displayName?.toLowerCase(), targetName.toLowerCase()) || namesMatch(entity.name, targetName)
+    }
+
+    /**
+     * Attempt pathfinding. If the Bot becomes 'stuck' then cancel pathfinding.
+     * The Bot is considered 'stuck' if it fails to move or perform mining/crafting/chest-interaction actions during a specified interval.
+     * @param {function()} pathFunc- a function utilizing pathfinder to move the Bot
+     * @param {object} [options={}] - optional parameters
+     * @param {number} [options.interval=5000] - how long in ms a Bot must be inactive to be considered 'stuck'
+     * @returns {Promise<boolean>} - true if pathing completes, or false if pathing is cancelled or otherwise interrupted
```

Comment:
remove hyphens between the param names and descriptions - the markdown generator adds them automatically, so this will look weird